packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2940,7 +2974,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         if (this._maxOpSizeInBytes >= 0) {
             // Chunking disabled
             if (!serializedContent || serializedContent.length <= this._maxOpSizeInBytes) {
-                return this.submitRuntimeMessage(type, content, batch, opMetadataInternal);
+                return this.submitRuntimeMessage(type,
+                                                 content,
+                                                 batch,
+                                                 serializedContent?.length ?? 0,
```

Comment:
I'm not sure if this is possible to fix in current state of main, but here is how chunking should work (chunking is disabled right now, mostly because it does not help when used in conjunction with batching):

1. Op content is compressed first (assuming it's over our limit of 2K)
2. Then we look at final size. If it's larger than chunk size (16Kb), then we break it into chunks.
    - Number of ops would be smaller, because we operate on much smaller compressed content.
Reverse on receiving ops.

Here is what we do now:
1. We Chunk ops bigger than 16K. Each op content is 16K
2. We compress each op separately. Each op becomes substantially smaller, but the efficiency of compression reduced.

Reply:
This comment actually caught that my current implementation would break chunked ops with compression enabled (the JSON can't be parsed since it's incomplete). I think it should be possible to implement the compression before the chunking as you described.

Reply:
This is a result of passing in the previously serialized content that doesn't include the chunking information (and as such there is no way to know which chunk is which when it comes back around). I'm looking to see if we can fix the passed in serialized value or if we need to go back to the way it was before.

Reply:
Is chunking enabled ?

Reply:
Figured it out, about to push a fix 😄